### ISSUE TITLE
Welcome page (initial webview)

### ISSUE THREAD

**Issue by deboer-tim** [containers/podman-desktop-extension-bootc#150]
### Is your enhancement related to a problem? Please describe

We're switching to use a webview instead of a quickpick for selecting the options to start a build. If we're using a webview anyway, we thought it would make a good place to start to explore bootable images and how to find the action.

### Describe the solution you'd like

Create a welcome/homepage webview for bootable containers. To start, it would provide an intro to bootable containers, tell the user how to get started, and link to documentation.

Sample for general design:
<img width="747" alt="Screenshot 2024-02-15 at 1 29 46 PM" src="https://github.com/containers/podman-desktop-extension-bootc/assets/19958075/d49e8b17-59ef-483c-9958-7612546b95ac">

**Comment by cdrage** [comment]
Updated the issue to reflect it'll be the "dashboard" as well and assigning it to myself as I've got some POC written

**Comment by cdrage** [comment]
Splitting this issue into two. This issue covers the "Welcome" section, issue #196 covers dashboard.

**Comment by cdrage** [comment]
@deboer-tim @slemeur 

#197 is in-review and feature complete.

We do not have UX/UI mockups, but it was talked about in previous calls that we would have the middle part of something like this:

![Screenshot 2024-03-20 at 7 40 10 PM](https://github.com/containers/podman-desktop-extension-bootc/assets/6422176/850022e4-214a-48bc-94ab-efe656781b4b)

Should we have it in the center when there are no builds completed?

![Screenshot 2024-03-20 at 7 40 50 PM](https://github.com/containers/podman-desktop-extension-bootc/assets/6422176/51cadfb9-67b9-47a0-8a3c-ac744fcae5c3)

This would make sense as if the user has done absolutely zero builds, the generic "no builds shown" should point to our README (the github page).

Let me know your thoughts.

**Comment by nichjones1** [comment]
Need a small mockup for “Welcome Page” for bootc that incorporates your “dashboard” for past X builds

**Comment by ekidneyrh** [comment]
I reused the Learning Center cards. Here's what that would look like: 

No builds:
![Screenshot from 2024-03-22 10-49-15](https://github.com/containers/podman-desktop-extension-bootc/assets/87311656/b7813c10-8241-47db-9d41-4dfb965f2c1b)

Builds with links visible:
![Screenshot from 2024-03-22 10-48-46](https://github.com/containers/podman-desktop-extension-bootc/assets/87311656/87c04b5a-2263-472d-b752-ea1d4205e658)

Builds with links collapsed:
![Screenshot from 2024-03-22 10-49-01](https://github.com/containers/podman-desktop-extension-bootc/assets/87311656/47e09938-1bdf-4a9a-8209-72cffd5b19c7)

Alt would be to have it stickied to the bottom of the screen, so if someone has a lot of builds it doesn't get lost amongst them:
![Screenshot from 2024-03-22 11-16-10](https://github.com/containers/podman-desktop-extension-bootc/assets/87311656/963fee43-4386-4e40-8887-979cb43fc789)

![Screenshot from 2024-03-22 11-18-06](https://github.com/containers/podman-desktop-extension-bootc/assets/87311656/7d051995-fb66-4e33-81f4-aa690bb37efd)

**Comment by cdrage** [comment]
> I reused the Learning Center cards. Here's what that would look like:
> 
> No builds: ![Screenshot from 2024-03-22 10-49-15](https://private-user-images.githubusercontent.com/87311656/315986531-b7813c10-8241-47db-9d41-4dfb965f2c1b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTExMTY0OTcsIm5iZiI6MTcxMTExNjE5NywicGF0aCI6Ii84NzMxMTY1Ni8zMTU5ODY1MzEtYjc4MTNjMTAtODI0MS00N2RiLTlkNDEtNGRmYjk2NWYyYzFiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzIyVDE0MDMxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTFjNzQ2MjA4MGE5MTFiYTcwNDBjNDk0NTM1MTI3ODdmNmY0ODZjODVjOTdkMTRkNDk0YTg1ODMzOTM4YTMyMzgmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.bRX7r_thjwns2my6g66F2B3QwNyGeqQdPNNP0XFiM24)
> 
> Builds with links visible: ![Screenshot from 2024-03-22 10-48-46](https://private-user-images.githubusercontent.com/87311656/315986541-87c04b5a-2263-472d-b752-ea1d4205e658.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTExMTY0OTcsIm5iZiI6MTcxMTExNjE5NywicGF0aCI6Ii84NzMxMTY1Ni8zMTU5ODY1NDEtODdjMDRiNWEtMjI2My00NzJkLWI3NTItZWExZDQyMDVlNjU4LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzIyVDE0MDMxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTJlNmZiMjU3ZDk0ZGQ0NWM5MzRlOGI2YWI1MTYwYmYzNTI0ODgxOWFkZTNmYjU5MzQ4NTEyYjIyMjRlNjY1ODQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.2lxobimf09vimoMdRDkLkRDzvU0-Wi_PMSBSO4T7ix8)
> 
> Builds with links collapsed: ![Screenshot from 2024-03-22 10-49-01](https://private-user-images.githubusercontent.com/87311656/315986539-47e09938-1bdf-4a9a-8209-72cffd5b19c7.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTExMTY0OTcsIm5iZiI6MTcxMTExNjE5NywicGF0aCI6Ii84NzMxMTY1Ni8zMTU5ODY1MzktNDdlMDk5MzgtMWJkZi00YTlhLTgyMDktNzJjZmZkNWIxOWM3LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzIyVDE0MDMxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPThkNjA0MjhlYTBjMjNiNTljZGYzYzM1Nzk2YWFhZDhjNzY3NDI5MmQ0YmZhZjViYWRkZDE2YTc1OTMyOTFkMDkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.eGzweHqKSJQQhr-RaR-fiVkX2QcMOdcmKBziLwGt4AI)
> 
> Alt would be to have it stickied to the bottom of the screen, so if someone has a lot of builds it doesn't get lost amongst them: ![Screenshot from 2024-03-22 11-16-10](https://private-user-images.githubusercontent.com/87311656/315988956-963fee43-4386-4e40-8887-979cb43fc789.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTExMTY0OTcsIm5iZiI6MTcxMTExNjE5NywicGF0aCI6Ii84NzMxMTY1Ni8zMTU5ODg5NTYtOTYzZmVlNDMtNDM4Ni00ZTQwLTg4ODctOTc5Y2I0M2ZjNzg5LnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzIyVDE0MDMxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWY2YzA5MTlmOGQ0YTVjMWMyMDNjYTBiMDkzMzdjZTJkZjJkYTc3NzdjMzgzODUyOGE2ZDdmZTUwNjMzNWVkNTkmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.pwG4P5o0Fu4erePF8gS1Wmss_GK2XTmPi6Y9CGUqfic)
> 
> ![Screenshot from 2024-03-22 11-18-06](https://private-user-images.githubusercontent.com/87311656/315989263-7d051995-fb66-4e33-81f4-aa690bb37efd.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MTExMTY0OTcsIm5iZiI6MTcxMTExNjE5NywicGF0aCI6Ii84NzMxMTY1Ni8zMTU5ODkyNjMtN2QwNTE5OTUtZmI2Ni00ZTMzLTgxZjQtYWE2OTBiYjM3ZWZkLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDAzMjIlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwMzIyVDE0MDMxN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWE1MzFjZTFiZmU4NWIxODE2NjAyYWM2YThmNzE1MzUzYzQwODZlYzkzNTgyOGUzZjc0Y2ExNmE3MmVlY2Y3OGImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.jxAcE4CL8-BqofhdWWon4hiZ5zOQZPBWyW9xz1Kwlrs)

IMHO I like the first one without the ability the toggle. It fills up the space, adds more information, and bluntly shows the user where the tutorials area (so we dont get questions asking how to build one.. and realize that they had forgotten to toggle it).

I like how if there are builds, it will show on the bottom with a toggle. However, I think we should add a little "X" on the top right and if you click it, it'll prompt you with a "Never show again?" and we can use a configuration setting to remember that.

WDYT @deboer-tim

**Comment by ekidneyrh** [comment]
> However, I think we should add a little "X" on the top right and if you click it, it'll prompt you with a "Never show again?" and we can use a configuration setting to remember that.

+1 I think this is a great idea

**Comment by cdrage** [comment]
> As a user, I'm not really understanding what that is. I want to be able to learn the basics, without having to leave the tool. In the page, we should have a minimal set of informations and descriptions about the extension (what it does, what it is, what thing to do first). Then we provide links to "learn more".
> 
> When it's empty, I should have a button "Run/Start Hello World bootable image" and it should be "doing/running" something in the application - not a redirect to an external page.

I agree and that having it within the tool would be beneficial without having to leave.

For the scope of this issue in the description: **To start, it would provide an intro to bootable containers, tell the user how to get started, and link to documentation.** the PR #212 will cover it.

To have the other parts implemented, we should open up separate issues.

1. Page that summarized bootc / explains minimally from within the tool, not an outside link.
2. Button that creates a bootable image. We should discuss how to do this as we'd have to explain the image is quite large (1.5GB average), what's in the image, progress bar of the image pulling, what the password is to login, etc.
3. What to do after?

LMK and feel free to open up any issues.

**Comment by deboer-tim** [comment]
The primary goal was to give the user some initial guidance on what bootable containers are and how to get started. I agree with Stevan that as a user this doesn't do the first and sends you externally for the second, which is not a great experience. I think we need an actual welcome screen here with some text and guidance, not just an 'empty screen'.

Nit: on the PD empty screens the big icon is in grey so that your eyes are drawn to the text, here it is really bright and does the opposite. On first start your eye should be drawn to what this extension is & next steps.

FWIW I'd prefer not to do a 'don't show me again' preference. In my experience users click on this when something is too in the way or annoying, and they are unlikely to ever find the preference and turn it on again. It would be better if the welcome or learning centre automatically reduced or got out of the way as you started, but still had a way to reopen.

**Comment by cdrage** [comment]
> The primary goal was to give the user some initial guidance on what bootable containers are and how to get started. I agree with Stevan that as a user this doesn't do the first and sends you externally for the second, which is not a great experience. I think we need an actual welcome screen here with some text and guidance, not just an 'empty screen'.
> 
> Nit: on the PD empty screens the big icon is in grey so that your eyes are drawn to the text, here it is really bright and does the opposite. On first start your eye should be drawn to what this extension is & next steps.
> 
> FWIW I'd prefer not to do a 'don't show me again' preference. In my experience users click on this when something is too in the way or annoying, and they are unlikely to ever find the preference and turn it on again. It would be better if the welcome or learning centre automatically reduced or got out of the way as you started, but still had a way to reopen.

You are right and having just the links like the mockup does not convey much information:

![Screenshot 2024-03-22 at 12 12 43 PM](https://github.com/containers/podman-desktop-extension-bootc/assets/6422176/46ba7cb5-54e7-4285-8edb-1bd8d5895be7)

as it's just links.

We could also just remove the top part and make the information "full screen" unless a build has been performed:

![Screenshot 2024-03-22 at 12 10 47 PM](https://github.com/containers/podman-desktop-extension-bootc/assets/6422176/9f03f138-adee-4d73-95cc-57e42cad5837)

Are there any reference screenshots of what it should look like that I can base the implementation on / what a full welcome screen would be?

**Comment by ekidneyrh** [comment]
The bright bootc image in the centre is just a placeholder until we get the final logo and can convert to a grey icon for this purpose. 

The cards with the links could contain more text / longer descriptions of useful information, expand to fill up the screen more and remove the 'Get Started' button for the time being?

**Comment by cdrage** [comment]
> The bright bootc image in the centre is just a placeholder until we get the final logo and can convert to a grey icon for this purpose.
> 
> The cards with the links could contain more text / longer descriptions of useful information, expand to fill up the screen more and remove the 'Get Started' button for the time being?

Had a quick discussion with @deboer-tim and we should try to fill more of the screen with a guide / more hand holding (even if it's a button that says "pull example image") so it's much easier to understand.

Let's all catch up this upcoming week with regards to creating a mockup.

**Comment by cdrage** [comment]
Result of meeting:

- Have it similar to the "empty" screens with the `podman run helloworld` commands, but have it with "Pull image" instead.
- Add a paragraph explaining the Bootable Container concept
- Keep it simple and just have test that says "learn more"
- No cards on the bottom
- Have it display / show when there are builds, but non-obtrusive.
